lib/Alchemy/Phrasea/Command/Maintenance/CleanUsersCommand.php

```diff
@@ -105,13 +123,48 @@ protected function doExecute(InputInterface $input, OutputInterface $output)
         }
         */
 
+        $cnx = $this->container->getApplicationBox()->get_connection();
+
+        $mode = $input->getOption('mode');
+
+        if (!in_array($mode, ['cleanup', 'purge'])) {
+            $output->writeln(sprintf("<error>bad \"mode\" value '%s'</error> (see possible values with --help)", $mode));
+
+            return 1;
+        }
+
         // --older_than
 
         $older_than = str_replace(['-', '/', ' '], '-', $input->getOption('older_than'));
-        if($older_than === "") {
+        if($mode == "cleanup" && $older_than === "" ) {
             $output->writeln("<error>set '--older_than' option.</error>");
             return 1;
         }
+
+        if ($mode == "purge") {
+            $sqlCountDeleted = "SELECT count(*) AS n FROM Users WHERE deleted = 1";
+            $stmt = $cnx->prepare($sqlCountDeleted);
+            $stmt->execute();
+            $nDeleted = $stmt->fetchColumn(0);
+            $stmt->closeCursor();
+
+            $helper = $this->getHelper('question');
+            $question = new ConfirmationQuestion(sprintf("Confirm deletion of %s deleted user(s) [y/n] : ", $nDeleted), false);
+
+            if (!$helper->ask($input, $output, $question)) {
+                return 0;
+            }
+
+            $sqlDelete = "DELETE FROM Users WHERE deleted = 1";
```

Comment:
normally, a "deleted=1" user should not have related data (basket, account, share, feedback, stories, log...) but we should delete (?) everything to be sure.
Q @nmaillat : purge will delete everything related ! maybe an an option `--new-owner=email` (or id) ?
We could also dynamically create a user "lost+found" ?